File: JavaScript/TwoSum.js
/** Returns the indices of two numbers in an array that add up to a target number.*/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

const twoSum = (nums, target) => {
  const map = new Map();
   
  for (let i = 0; i < nums.length; i++) { 
    const complement = target - nums[i];
    
    if (map.has(complement)) {
      return [map.get(complement), i];
    }
    
    map.set(nums[i], i);
  }
  
  return [];
};

File: TwoSum.py
# Returns the indices of two numbers in an array that add up to a target number.

class Solution:
    def twoSum(self, nums, target):
        num_to_index = {}
        
        for i, num in enumerate(nums):
            complement = target - num
            
            if complement in num_to_index:
                return [num_to_index[complement], i]
            
            num_to_index[num] = i
        
        return []
